### PR TITLE
process: update Node.js sec membership policy

### PR DESCRIPTION
**Pull request by sam-github**
The policy was spread across two files, had mild contradictions (one doc
    said membership was confirmed by the TSC, the other said by it was
    confirmed by the current members), referenced the unused github issue
    tracker (HackerOne is used now), and had lots of mention of the
    @nodejs/security team, which does not in itself have access to ANY
    private issues or patches (I removed those refs).
    
    The policy is essentially unchanged, just more readable, and in one file
    so it shouldn't drift out of sync as easily.

### CODE REVIEW

File: processes/security_team_members.md [hunk 1]
```diff
@@ -64,6 +72,8 @@ List is from ["security" alias](https://github.com/nodejs/email/blob/master/iojs
 
 List is from [nodejs/teams/security](https://github.com/orgs/nodejs/teams/security/members).
 
+XXX Why isn't the list from nodejs-private/teams/security? That's the actual one.  Or maybe they are supposed to be synced.
```
Comment (Trott) [inline processes/security_team_members.md]: Pretty sure they are supposed to be sync'ed.

File: processes/security_team_members.md [hunk 2]
```diff
@@ -13,27 +13,35 @@ and must be approved by current team members.
...
 
 - @bnoordhuis - **Ben Noordhuis**
 - @jasnell - **James M Snell**
 - @shigeki - **Shigeki Ohtsu**
 
-List is from ["security" alias](https://github.com/nodejs/email/blob/master/iojs.org/aliases.json).
 
 ## Team with access to security issues
 
+XXX Unclear what meaning of this is now that we use H1. Almost certainly not up to date.
```
Comment (Trott) [inline processes/security_team_members.md]: +1 for removing it. It was probably useful when we were first figuring things out, but yeah, keeping multiple lists in multiple places is going to pretty much guarantee stuff gets out of sync at some point and then you don't know which one is the source of truth.

File: processes/security_team_members.md [hunk 3]
```diff
@@ -98,3 +108,5 @@ who have access to
 
 Every member of the team with access to security issues should have access to
 the private security patches as well.
+
+XXX ... But they don't, and if they did, we wouldn't have to put the list in twice.
```
Comment (Trott) [inline processes/security_team_members.md]: I'm not sure what you mean here. If you are in @nodejs-private/security, you have access. (Or am I misunderstanding how something works?)

IIUC, @nodejs/security is basically legacy except that it's useful for @-notifications in repos in the @nodejs org.

Comment (sam-github) [reply processes/security_team_members.md]: There are two sections here, one above for "issues" and this one for "patches". First is doced as being the people in @nodejs/security, the second is doced as org/nodejs-private/people... but then it says everyone in @nodejs/security should be in nodejs-private/people... so they are basically the same lists.  And also, the "issues" is https://github.com/nodejs-private/security and the patches are https://github.com/nodejs-private/node-private, and it happens that BOTH issues and patches have exactly the same team: https://github.com/orgs/nodejs-private/teams/security. All of which is to say, describing them as having two different lists of members is not correct.

Also, the issues have been archived... so its not even relevant. Anyhow, these are more notes for myself, I'll do a second pass and clean this up.

**Comment by sam-github** [comment]
@nodejs/tsc @nodejs/security There is something I'm not clear on, is everyone who has access to the Node.js program in H1 on the Triage team? All those people have the _ability_ to triage issues, but I'm not sure that you all _will_  triage issues, and _want_ to triage issues. A number of the TSC members may have access to H1 because of a responsibility to be aware of activity, but not out of an intention to be actively involved. Does it even matter? Maybe not, but if none of the "triage team" is actually triaging, things will go badly. Then again, perhaps that's best dealt with in the weekly TSC meetings, if there are open vulns that are not getting dealt with.

I bother bringing it up because the old triage team was small, and with H1, its now quite large, and its not obvious that the everyone who accepted the H1 invitation realized they were implicitly joining the triage team.

I'm a big fan of simplifying the complexity of the team structure. It does appear there are unavoidably, at this moment, 3 actual resources, and corresponding lists of people with access:
- H1's nodejs program
- https://github.com/nodejs-private/node-private, where we prepare patches that cannot be seen in the public yet, which is larger than the H1 list because it includes releasers
- @nodejs/security, used to attract people's attention on github (and yes, I'm tagging you now on purpose)

It is possible that we can force the latter two github lists to have the same members (manually, or automatically, don't we have some tools somewhere for updating markdown membership lists from github APIs?), in which case, we only have to document two lists: triage team and security team.

What should we do here?

**Comment by sam-github** [comment]
and to further flood mailboxes, sorry:

> realized they were implicitly joining the triage team.

The triage team policy used to be that everyone individually PRed their name here, indicating acceptance of the privacy policy. I'm not sure that happened for the TSC when they got H1 access. I'm OK with removing the process, or following the process, but we should probably do one or the other.

**Comment by richardlau** [comment]
> It is possible that we can force the latter two github lists to have the same members (manually, or automatically, don't we have some tools somewhere for updating markdown membership lists from github APIs?), in which case, we only have to document two lists: triage team and security team.
> 

Yes: https://github.com/nodejs/node-core-utils/blob/master/docs/ncu-team.md

**Comment by sam-github** [comment]
PTAL, now lists only the two teams, the one with access to H1, and the one with access to nodejs-private/node-private (updated to use ncu-team for keeping synced).

@Trott @mhdawson

**Comment by Trott** [comment]
> PTAL, now lists only the two teams, the one with access to H1, and the one with access to nodejs-private/node-private (updated to use ncu-team for keeping synced).
> 
> @Trott @mhdawson

Taking a quick look, seems OK to me. I'm not on Security WG so I'll leave it to actual members to green-check this, but 👍.

**Comment by sam-github** [comment]
@Trott  You are among the correct people to review this, IMO. While the process wording needs updating because triage moved to H1, it is the sole responsibility of the TSC to determines the membership of the Node.js triage team, see https://github.com/nodejs/security-wg/blob/master/processes/security_team_membership_policy.md

@nodejs/TSC, PTAL

**Comment by sam-github** [comment]
> While the process wording needs updating because triage moved to H1

... I updated the process wording, by moving its text into the same file as the membership privacy policy/membership list. So, that's all cleaned up now.

**Comment by sam-github** [comment]
@nodejs/tsc, PTAL - it is TSC's preexisting responsiblity to appoint sec team members, I updated the lists of members, and made minor changes to the policy falling out of the HackerOne use, which moved the location of issues out of github (so access to it is not controlled any longer by a github team).

File: processes/security_team_members.md [hunk 4]
```diff
@@ -7,94 +7,74 @@ yet been disclosed publicly, including the existence of issues, expectations of
...
-- @shigeki - **Shigeki Ohtsu**
+## Team that triages security reports against Node.js
 
-List is from ["security" alias](https://github.com/nodejs/email/blob/master/iojs.org/aliases.json).
+The [TSC](https://github.com/nodejs/node#tsc-technical-steering-committee)
+are all members of the Triage Team.
 
-## Team with access to security issues
+These non-TSC and TSC Emeriti are Triage Team members:
+- [bnoordhuis](https://github.com/bnoordhuis) - **Ben Noordhuis**
```
Comment (Trott) [inline processes/security_team_members.md]: The mixture of `-` and `*` causes GitHub to render the list with gaps. 

Screen Shot 2019-07-08 at 1 18 35 PM

```suggestion
* [bnoordhuis](https://github.com/bnoordhuis) - **Ben Noordhuis**
```

File: processes/security_team_members.md [hunk 5]
```diff
@@ -7,94 +7,74 @@ yet been disclosed publicly, including the existence of issues, expectations of
...
+* [indutny](https://github.com/indutny) - **Fedor Indutny**
+* [rvagg](https://github.com/rvagg) - **Rod Vagg**
+- [vdeturckheim](https://github.com/vdeturckheim) - **Vladimir de Turckheim**
```
Comment (Trott) [inline processes/security_team_members.md]:
```suggestion
* [vdeturckheim](https://github.com/vdeturckheim) - **Vladimir de Turckheim**
```

**Comment by mhdawson** [comment]
Some background I think that is relevant:

1) The suggestion was made to obsolete the private github repo that was used to manage discussion of vulnerabilities to force all discussion to H1
2) All TSC members previously had access to the private github repo
3) All TSC members were therefore invited to H1 so that they'd continue to have insight into the discussions on vulnerabilities.

I don't think there was any intention to change the composition of the triage team by adding the TSC members to H1.  ie it still makes sense to have people PR themselves into a list that identifies who the core security triage team is.

File: processes/security_team_members.md [hunk 6]
```diff
@@ -7,94 +7,74 @@ yet been disclosed publicly, including the existence of issues, expectations of
...
 
-### Emeritus
+Membership on the security teams can be requested via an issue in the TSC repo.
 
-- @jasnell - **James M Snell**
-- @shigeki - **Shigeki Ohtsu**
+## Team that triages security reports against Node.js
 
-List is from ["security" alias](https://github.com/nodejs/email/blob/master/iojs.org/aliases.json).
+The [TSC](https://github.com/nodejs/node#tsc-technical-steering-committee)
```
Comment (mhdawson) [inline processes/security_team_members.md]: I don't believe this is accurate.  All TSC members have access  to info about discussions/patches related to security vulnerabilities but that does not mean they are part of the team that has volunteered to do the triage.

**Comment by sam-github** [comment]
@mhdawson I am also unclear about who is a member of the triage team, see my questions here: https://github.com/nodejs/security-wg/pull/558#issuecomment-508870881

The triage team was a mailing list of people who received notification of new vuln reports. That mailing list no longer exists.

With H1, everyone who is a member of the H1 program gets notifications, but I agree, its not clear which (if any) of them is on the triage team.

I'd add an extra section specific to triage, but I don't know who the current team is. Perhaps a re-call for volunteers should be made at the next TSC meeting?

**Comment by mhdawson** [comment]
I agree re-validating who is currently on the core triage team would make sense.  

@mcollina, @MylesBorins, @jasnell as well as yourself are the people who I've seen doing that work recently (apologies  in advance to those that I've missed as this is just based on my recent memory of activity notifications I've gotten for H1).

Maybe start with this list and then confirm those members and ask if we've missed anybody?  I suspect @mcollina, @MylesBorins, @jasnell will be able to fill in the gaps.

**Comment by mcollina** [comment]
Everybody on the TSC has access to H1 because it’s there that the main conversation happen about vulnerabilities. Initial triaging is facilitates by H1 staff, so it’s not a burden.

The current problem is a) deciding what to do for the borderline cases and b) fixing protocol-relates issues (http, http2, etc). Some of those are extremely hard, and there are not enough people engaged in the process.

**Comment by sam-github** [comment]
@mhdawson See @mcollina 's comment above, H1 is responsible for initial triage, something node used to do itself. I doesn't appear we need an initial triage team anymore.

**Comment by sam-github** [comment]
@mhdawson PTAL. Triage team is restored to its previous value.

**Comment by sam-github** [comment]
FTR, removing @mhdawson from triage, he shouldn't have been there, he only had access to H1. See https://github.com/nodejs/security-wg/pull/414#discussion_r225695620

**Pull request merged**